src/main/java/org/dcsa/tnt/service/impl/TNTEventSubscriptionTOServiceImpl.java

```diff
@@ -0,0 +1,357 @@
+            return eventSubscription;
+          };
+
+  @Override
+  public Mono<TNTEventSubscriptionTO> update(TNTEventSubscriptionTO eventSubscriptionTO) {
+    return eventSubscriptionRepository
+        .deleteEventTypesForSubscription(eventSubscriptionTO.getSubscriptionID())
+        .thenReturn(eventSubscriptionTO)
+        .map(eventSubscriptionTOToEventSubscription)
+        .flatMap(
+            updated ->
+                eventSubscriptionService
+                    .findById(updated.getSubscriptionID())
+                    .map(
+                        original -> {
+                          updated.setSecret(original.getSecret());
+                          updated.copyInternalFieldsFrom(original);
+                          return updated;
+                        }))
+        .flatMap(eventSubscriptionService::update)
+        .flatMap(ignored -> createEventTypes(eventSubscriptionTO));
+  }
+
+  private Mono<TNTEventSubscriptionTO> createEventTypes(
+      TNTEventSubscriptionTO eventSubscriptionTO) {
+
+    List<EventType> eventTypes;
+
+    if (CollectionUtils.isEmpty(eventSubscriptionTO.getEventType())) {
+      eventTypes = ALL_ALLOWED_EVENT_TYPES;
+      eventSubscriptionTO.setEventType(eventTypes);
+    } else {
+      eventTypes = eventSubscriptionTO.getEventType();
+    }
+
+    return Flux.fromIterable(eventTypes)
+        .concatMap(
+            eventType ->
+                eventSubscriptionRepository.insertEventTypeForSubscription(
+                    eventSubscriptionTO.getSubscriptionID(), eventType))
+        .then(Mono.just(eventSubscriptionTO));
+  }
+
+  private Mono<Void> createTransportDocumentEventTypes(TNTEventSubscriptionTO eventSubscriptionTO) {
+
+    if (null == eventSubscriptionTO.getTransportDocumentTypeCode()) {
+      eventSubscriptionTO.setTransportDocumentTypeCode(ALL_TRANSPORT_DOCUMENT_TYPES);
+    }
+
+    List<TransportDocumentTypeCode> transportDocumentTypeCodes =
+        eventSubscriptionTO.getTransportDocumentTypeCode();
+    return Flux.fromIterable(transportDocumentTypeCodes)
+        .flatMap(
+            td ->
+                eventSubscriptionRepository.insertTransportDocumentEventTypeForSubscription(
+                    eventSubscriptionTO.getSubscriptionID(), td))
+        .then();
+  }
+
+  private Mono<Void> createShipmentEventType(TNTEventSubscriptionTO eventSubscriptionTO) {
+
+    if (null == eventSubscriptionTO.getShipmentEventTypeCode()) {
+      eventSubscriptionTO.setShipmentEventTypeCode(ALL_SHIPMENT_EVENT_TYPES);
+    }
+
+    List<ShipmentEventTypeCode> shipmentEventTypeCode =
+        eventSubscriptionTO.getShipmentEventTypeCode();
+
+    return Flux.fromIterable(shipmentEventTypeCode)
+        .flatMap(
+            s ->
+                eventSubscriptionRepository.insertShipmentEventTypeForSubscription(
+                    eventSubscriptionTO.getSubscriptionID(), s))
+        .then();
+  }
```

Comment:
Even though this is not needed in OVS - it is needed in Documentation_Hub - I think it belongs to EventCore